queries/nvim/highlights.scm

```diff
@@ -16,14 +16,47 @@
   "then"
   "when"
 ] @keyword
-["def" "fun"] @keyword.function
-["class" "enum" "module" "struct" "type" "union"] @keyword.type
-["require"] @keyword.import
-["return" "yield"] @keyword.return
-["if" "else" "elsif" "unless"] @keyword.conditional
-["while" "until"] @keyword.repeat
-["rescue"] @keyword.exception
-[(private) (protected) "abstract"] @keyword.modifier
+
+[
+  "def"
+  "fun"
+] @keyword.function
+
+[
+  "class"
+  "enum"
+  "module"
+  "struct"
+  "type"
+  "union"
+] @keyword.type
+
+"require" @keyword.import
+
+[
+  "return"
+  "yield"
```

Comment:
The documentation for this highlight group specifically suggests `yield`

https://github.com/neovim/neovim/blob/7e1c1ff7fcf2cbc564c90a656124b70ad8bb4d5f/runtime/doc/treesitter.txt#L427

I think this makes sense, since `yield` is "returning" into the yielded block. If that block calls `break`, then control won't continue past yield